usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateAgencyLangAndFeedInfoFeedLangMatch.java

```diff
@@ -0,0 +1,94 @@
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsDataRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Use case to validate that `agency.agency_lang` and `feed_info.feed_lang` match. This use case is triggered after
+ * completing the {@code GtfsDataRepository} provided in the constructor with {@code FeedInfo} entities.
+ */
+public class ValidateAgencyLangAndFeedInfoFeedLangMatch {
+    private static final String MUL = "mul";
+    private final GtfsDataRepository dataRepo;
+    private final ValidationResultRepository resultRepo;
+    private final Logger logger;
+
+    /**
+     * @param dataRepo   a repository storing the data of a GTFS dataset
+     * @param resultRepo a repository storing information about the validation process
+     * @param logger     a logger to log information about the validation process
+     */
+    public ValidateAgencyLangAndFeedInfoFeedLangMatch(final GtfsDataRepository dataRepo,
+                                                      final ValidationResultRepository resultRepo,
+                                                      final Logger logger) {
+        this.dataRepo = dataRepo;
+        this.resultRepo = resultRepo;
+        this.logger = logger;
+    }
+
+    /**
+     * Use case execution method: checks if for each {@code Agency} defines a value for `agency.agency_lang` that
+     * matches the value contained in `feed_info.feed_lang` if `feed_info.txt` is provided. If this requirement is not
+     * met, a {@code FeedInfoLangAgencyLangMismatchNotice} is added to the {@code ValidationResultRepo} provided
+     * in the constructor.
+     */
+    public void execute() {
+        logger.info("Validating rule 'E055 - Mismatching feed and agency language fields'");
+
+        if (dataRepo.getFeedInfoAll().size() == 0) {
+            return;
+        }
+        // .get can be used without isPresent check here since line 55 ensures the presence of at least 1 element
+        // in the map
+        final String feedInfoFeedLang =
+                dataRepo.getFeedInfoAll().values().stream().findFirst().get().getFeedLang();
+        final Set<String> agencyLangCollection = new HashSet<>();
+        dataRepo.getAgencyAll().forEach((agencyId, agency) -> agencyLangCollection.add(agency.getAgencyLang()));
+        if (feedInfoFeedLang.equals(MUL)) {
+            // If feed_lang is mul and there isn't more than one agency_lang, that's an error
+            if (agencyLangCollection.size() <= 1) {
+                resultRepo.addNotice(new FeedInfoLangAgencyLangMismatchNotice(agencyLangCollection));
+                return;
+            }
+            return;
+        }
+        // If there is more than one agency_lang and feed_lang isn't mul, that's an error
+        if (agencyLangCollection.size() > 1 && !agencyLangCollection.contains(feedInfoFeedLang)) {
```

Comment:
Sorry, just catching this - I believe `!agencyLangCollection.contains(feedInfoFeedLang)` should be removed from this IF statement. It should be an error regardless if at least one of the agency langs matches the feed lang, because in all cases when there is more than one agency lang the feed lang should be `mul`.

Reply:
What happens is that with the current implementation, the following situation would not be caught:
- `feed_info..feed_lang = `US` and `agency.txt` defines `FR` 

which leads tests `feedLangNotMulAndOnlyOneMismatchingAgencyLangShouldGenerateNotice` ans `nonMatchingFeedInfoFeedLandShouldGenerateNotice` to fail

So I cam to the conclusion that we needed an additional check to pass handle these situations and pass these tests.

Reply:
Nice! That's what unit tests are for :).

Reply:
Sorry I misunderstood something. I discussed, yes, the code will be changed.